homeassistant/components/alarm_control_panel/egardia.py

```diff
@@ -94,18 +94,13 @@ def lookupstatusfromcode(self, statuscode):
         status = 'UNKNOWN'
         if self._rs_codes is not None:
```

Comment:
If you make sure self._rs_codes is always a dict, ie at least an empty dict `{}`, you can remove this check and just iterate below. You can add a default empty dict to the config validation schema in the egardia component:
```py
vol.Optional(CONF_REPORT_SERVER_CODES, default={}): SERVER_CODE_SCHEMA,
```

Reply:
thanks, updated.